src/public/manageDomain.ts

```diff
@@ -0,0 +1,173 @@
+const envList = helper.getElement('.envList')
+const environmentVariables: EnvironmentItem[] = []
+const fullDomain = window.location.pathname.split('/').pop()
+document.getElementById('full-domain').innerText = fullDomain
+
+class EnvironmentItem {
+  name: string
+  value: string
+  isValid = false
+
+  private nameInputElement: HTMLInputElement
+  private valueInputElement: HTMLInputElement
+
+  private readonly IS_INVALID = 'is-invalid'
+  private readonly LETTER_NUMBER_REGEX = /^[A-Z0-9_]+$/g
+  private readonly ELEMENT_HTML = `
+    <div class="input-group">
+      <input 
+        type="text"
+        aria-label="Name"
+        class="form-control text-uppercase text-monospace name-input ${this.IS_INVALID}"
+        placeholder="NAME"
+      >
+      <div class="input-group-prepend input-group-append">
+        <span class="input-group-text">=</span>
+      </div>
+      <input
+        type="text"
+        aria-label="Value"
+        class="form-control text-monospace value-input"
+        placeholder="Value"
+      >
+      <div class="input-group-append">
+        <button
+          class="btn btn-outline-danger remove-button"
+          type="button"
+        >
+          Remove
+        </button>
+      </div>
+      <div class="invalid-feedback">
+        Name must contain only letters, number, or underscore.
+      </div>
+    </div>
+  `
+
+  constructor(name?: string, value?: string) {
+    this.name = name
+    this.value = value
+    this.createElement()
+    if (name && value) {
+      this.setInputValues()
+    }
+    this.nameInputElement.focus()
+  }
+
+  /**
+   * Create the list element and add the event listeners for inputs and button
+   */
+  private createElement(): void {
+    const environmentElement = document.createElement('li')
+    environmentElement.innerHTML = this.ELEMENT_HTML
+    environmentElement.classList.add(
+      'list-group-item',
+      'd-flex',
+      'align-items-center'
+    )
+    envList.appendChild(environmentElement)
+    this.nameInputElement = environmentElement.querySelector('.name-input')
+    this.valueInputElement = environmentElement.querySelector('.value-input')
+    this.nameInputElement.addEventListener('input', () =>
```

Comment:
I would probably move the event listeners outside the creat element method. Unless you think it makes more sense here.

```
createAndAppendElements()
setupEventListeners() || handleEvents()
```

Reply:
You could also create a private method for each event handler making it even more modular.

Reply:
Only the name needs some checking when it's input, so a private method makes sense. For the value the event is a simple value assignment, I don't think it needs a specific class method. If this changes in the future it can be refactored.